main.go

```diff
@@ -179,6 +180,26 @@ func main() {
 		os.Exit(1)
 	}
 
+	var autoAnnotationConfig auto.AnnotationConfig
+	var autoAnnotation *auto.AnnotationMutators
+	if os.Getenv("DISABLE_AUTO_ANNOTATION") != "true" {
+		if err := json.Unmarshal([]byte(autoAnnotationConfigStr), &autoAnnotationConfig); err != nil {
+			setupLog.Error(err, "unable to unmarshal auto-annotation config")
+		} else {
+			autoAnnotation = auto.NewAnnotationMutators(
+				mgr.GetClient(),
+				mgr.GetAPIReader(),
```

Comment:
What does this get us? Is it to bypass the cache? Do we have a usecase for that?

Reply:
It's to bypass the cache. During testing, we ran into a similar issue to https://github.com/jaegertracing/jaeger-operator/issues/809 where the cache wasn't available for reads yet.